Lib/getpass.py

```diff
@@ -151,6 +175,33 @@ def _raw_input(prompt="", stream=None, input=None):
     return line
 
 
+def _input_with_echochar(prompt, stream, input, echochar):
+    if not stream:
+        stream = sys.stderr
+    if not input:
+        input = sys.stdin
+    prompt = str(prompt)
+    stream.write(prompt)
+    stream.flush()
+    passwd = ""
+    while True:
+        char = input.read(1)
+        if char == '\n' or char == '\r':
+            break
+        if char == '\x03':
+            raise KeyboardInterrupt
+        if char == '\x7f' or char == '\b':
```

Comment:
`\x7f` is the DEL character, but AFAIR, `DEL` deletes the character on the *right*, not the left. So it shouldn't matter here? (`\b` is backspace so it's fine though).

Reply:
yea, you're right. I'll remove the `\x7f` char then.
anyway we don't handle this char in Windows implementation

Reply:
as far as i know the character that will be sent depends on the context. for example when i press Backspace on my keyboard i get `\x07` so i will leave handle of this char

Reply:
Ah right, actually Wikipedia tells me that:

> Terminal emulators may produce DEL when ← Backspace key or Control+← Backspace or Control+? are typed.